File: tests/syntax/supported/elixir/highlight.ex
# Constants and helpers
defmodule Calculator do
  @moduledoc "A tiny calculator module."

  @pi 3.14159

  def square(x) do
    x * x
  end

  def area(radius) do
    @pi * square(radius)
  end

  def classify(n) when is_integer(n) do
    cond do
      n < 0 -> :negative
      n == 0 -> :zero
      true -> :positive
    end
  end

  def shout(name) do
    name
    |> String.upcase()
    |> Kernel.<>("!")
  end
end

flag = true
greeting = "Hello, world"
IO.puts(greeting)

File: tests/syntax/supported/erlang/highlight.erl
%% Constants and helpers
-module(rotide_highlight).
-export([start/0, classify/1]).
-record(user, {name, active = true}).

-spec start() -> ok.
start() ->
    User = #user{name = <<"Ada">>},
    case classify(42) of
        positive ->
            io:format("~p~n", [User#user.name]),
            ok;
        Other ->
            {error, Other}
    end.

classify(N) when is_integer(N), N > 0 ->
    positive;
classify(0) ->
    zero;
classify(_) ->
    negative.
